compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/ObservabilityJMethodCall.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.ballerinalang.compiler.bir.codegen.interop;
+
+import io.ballerina.tools.diagnostics.Location;
+import org.wso2.ballerinalang.compiler.bir.model.BIRNonTerminator;
+import org.wso2.ballerinalang.compiler.bir.model.BIROperand;
+import org.wso2.ballerinalang.compiler.bir.model.BIRVisitor;
+import org.wso2.ballerinalang.compiler.bir.model.InstructionKind;
+
+import java.util.List;
+
+/**
+ * Java method invocation call modeled as BIR NonTerminator Instruction.
+ *
+ * @since 2201.4.0
+ */
+public class ObservabilityJMethodCall extends BIRNonTerminator {
```

Comment:
Since we have `JInstruction` implementation for java specific instructions, shouldn't we extend this from `JInstruction`? 
In that way, we can get rid of the `instanceOf` checks and use the `jKind` field. 
https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JInstruction.java

Reply:
+1. Also shall we use general name for class. ex. JMethodCallInstruction